### PR TITLE
Music Rework

### PR DESCRIPTION
**Pull request by QuantumlyTangled**
### Description of Your Pull Request
This PR enchances and simplifies the Music Player

### Changes Proposed and or Files Changed
Loop, Soundcloud and Spotify support will be coming in a future PR called 
> Advance Music Rework
### Label Your Pull Request
- [x] This PR modifies existing code to enhance the bot
- [ ] This PR fixes a bug within the bot
- [x] This PR removes/renames methods or properties
- [x] This PR adds methods, features, commands or properties

### CODE REVIEW

File: lib/music/LavaManager.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+const { PlayerManager } = require("discord.js-lavalink");
+
+class LavaManager {
+
+    constructor(client) {
+        this.client = client;
+    }
+
+    manager() {
```
Comment (MrJacz) [inline lib/music/LavaManager.js]: The class should extend PlayerManager

File: lib/music/MusicManager.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
+const LavaManager = require("./LavaManager.js");
+const { Collection } = require("discord.js");
+
+class MusicManager extends LavaManager {
```
Comment (MrJacz) [inline lib/music/MusicManager.js]: The entire class except region stuff makes no sense to me.

**Comment by MrJacz** [comment]
This is what i suggested in discord.

File: lib/music/MusicManager.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+const LavaManager = require("./LavaManager.js");
+
+class MusicManager extends Collection {
+
+    constructor(client, options = {}) {
+        super();
+        this.client = client;
+        this.options = options;
+
+        this.manager = new LavaManager(this.client, this.options);
```
Comment (MrJacz) [inline lib/music/MusicManager.js]: The LavaManager should be attached to client

**Comment by MrJacz** [comment]
I've noticed there is a lib folder in this PR, you might want to move PenguClient and various things into there, but it should be something like
`/lib/structures/...`
`/lib/extensions/...` and so on

File: lib/structures/MusicCommand.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
+const { Command } = require("klasa");
+
+class MusicCommand extends Command {
+
+    constructor(client, store, file, core, options = {}) {
+        // By nature, music commands only run in VoiceChannels, which are in Guilds.
+        options.runIn = "runIn" in options ? options.runIn : ["text"];
```
Comment (MrJacz) [inline lib/structures/MusicCommand.js]: this should be `this.runIn` and under super

File: lib/structures/MusicCommand.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+        super(client, store, file, core, options);
+
+        /**
+		 * Whether this command requires an active VoiceConnection or not
```
Comment (MrJacz) [inline lib/structures/MusicCommand.js]: Fix the JSDoc please

File: lib/music/Queue.js [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+        this.id = data.info.identifier;
+        this.seekable = data.info.isSeekable;
+        this.author = data.info.author;
+        this.time = data.info.isStream ? 0 : data.info.length;
+        this.stream = data.info.isStream;
+        this.position = data.info.position;
+        this.skips = new Set();
+    }
+
+    get secondafy() {
```
Comment (MrJacz) [inline lib/music/Queue.js]: friendlyDuration it should be called

File: lib/music/Queue.js [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+    constructor(data, requester) {
+        Object.defineProperty(this, "data", { value: data });
+        this.requester = requester;
+        this.track = data.track;
+        this.title = data.info.title;
+        this.url = data.info.uri;
+        this.id = data.info.identifier;
+        this.seekable = data.info.isSeekable;
+        this.author = data.info.author;
+        this.time = data.info.isStream ? 0 : data.info.length;
```
Comment (MrJacz) [inline lib/music/Queue.js]: duration not time

File: lib/music/Queue.js [hunk 8]
```diff
@@ -0,0 +1,49 @@
+const { showSeconds } = require("../Util/Util");
+class Queue {
```
Comment (MrJacz) [inline lib/music/Queue.js]: `Song` not `Queue`

File: lib/music/Queue.js [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+        this.stream = data.info.isStream;
+        this.position = data.info.position;
+        this.skips = new Set();
+    }
+
+    get secondafy() {
+        return showSeconds(this.time);
+    }
+
+    secondafyAnything(passed) {
```
Comment (MrJacz) [inline lib/music/Queue.js]: this is pointless

File: lib/music/Queue.js [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+        return showSeconds(passed);
+    }
+
+    toString() {
+        return `${this.title} (${this.url})`;
+    }
+
+    toJSON() {
```
Comment (MrJacz) [inline lib/music/Queue.js]: this is unneeded

File: lib/extensions/PenguGuild.js [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+
+module.exports = Structures.extend("Guild", Guild => {
+    /**
+	 * Pengu's Extended Guild
+	 * @extends {Guild}
+	 */
+    class PenguGuild extends Guild {
+
+        /**
+		 * @param {...Object} args Normal D.JS guild args
```
Comment (MrJacz) [inline lib/extensions/PenguGuild.js]: this should be `...*`

File: lib/extensions/PenguGuild.js [hunk 12]
```diff
@@ -0,0 +1,20 @@
...
+		 */
+		constructor(...args) { // eslint-disable-line
```
Comment (MrJacz) [inline lib/extensions/PenguGuild.js]: remove the disable line and fix indent

File: lib/Util/Util.js [hunk 13]
```diff
@@ -0,0 +1,22 @@
...
+        throw new Error(`The ${this.constructor.name} class may not be instantiated.`);
+    }
+
+    /**
+	 * Turns a time string into a formated one
+	 * @param {string} duration the inputed duration to format
+	 * @param {string} [format] The format to use
+	 * @returns {string}
+	 */
+    static showSeconds(duration, format = "h:mm:ss") {
```
Comment (MrJacz) [inline lib/Util/Util.js]: this should be
```js
static showSeconds(ms) {
    const sec = Math.floor((ms / 1000) % 60).toString();
    const min = Math.floor((ms / (1000 * 60)) % 60).toString();
    const hrs = Math.floor(ms / (1000 * 60 * 60)).toString();
    return `​${hrs.padStart(2, "0")}:${min.padStart(2, "0")}:${sec.padStart(2, "0")}`​;
}
```

File: lib/extensions/PenguGuild.js (hunk 12 above)
Comment (QuantumlyTangled) [reply lib/extensions/PenguGuild.js]: Again the indent is just github since in the editor it looks fine

**Comment by QuantumlyTangled** [comment]
And Util/Util.js wasnt taken from sneyra it is a klasified version of Sistines second parser which was in moment

**Comment by MrJacz** [comment]
Well, its still not correct and should be the requested code

**Comment by QuantumlyTangled** [comment]
That is added as you requested in the last commit

File: lib/Util/Util.js [hunk 14]
```diff
@@ -0,0 +1,29 @@
...
+        throw new Error(`The ${this.constructor.name} class may not be instantiated.`);
+    }
+
+    /**
+	 * Turns a time string into a formated one
+	 * @param {string} timeString the inputed duration to format
+	 * @param {string} [format] The format to use
+	 * @returns {string}
+	 */
+    static formatTime(timeString, format = "h:mm:ss") {
```
[the remark this answers is not in the document]
Comment (QuantumlyTangled) [reply lib/Util/Util.js]: The kyra has been answered to dear Adi please ask them for further info

**Comment by QuantumlyTangled** [comment]
You correct everything yet didnt see that its MusicInteface and not MusicInterface, granted I did not see it my self but still

File: extendables/music.js [hunk 15]
```diff
@@ -0,0 +1,13 @@
+const { Extendable } = require("klasa");
```
Comment (MrJacz) [inline extendables/music.js]: no... music should be a property of guild not a getter

File: extendables/player.js [hunk 16]
```diff
@@ -0,0 +1,13 @@
+const { Extendable } = require("klasa");
```
Comment (MrJacz) [inline extendables/player.js]: 1st) this shouldn't be attached to guild
2nd) if we needed this it should be a Structures.extend

File: lib/music/MusicManager.js [hunk 17]
```diff
@@ -1,18 +1,36 @@
-const { Collection, Client } = require("discord.js"); // eslint-disable-line no-unused-vars
-const LavalinkClient = require("./LavalinkClient.js");
+const Discord = require("discord.js");
```
Comment (MrJacz) [inline lib/music/MusicManager.js]: yet, this should be removed and music interface should just be used

Comment (QuantumlyTangled) [reply lib/music/MusicManager.js]: Lavalink client it attached to client as well as this

File: lib/music/MusicInterface.js [hunk 18]
```diff
@@ -0,0 +1,219 @@
...
+     * @since 2.0.1
+	 * @param {Discord.voiceChannel} voiceChannel The voice channel the player should join
+	 * @returns {Object<Player>}
+	 */
+    join(voiceChannel) {
+        return this.client.lavalink.join({
+            guild: this.guild.id,
+            channel: voiceChannel.id,
+            host: this.client.lavalink.getIdealHost(this.guild.region)
+        }, { selfdeaf: this.guild.configs.selfDeaf ? this.guild.configs.selfDeaf : true });
```
Comment (MrJacz) [inline lib/music/MusicInterface.js]: this shouldn't be a config, it should always be true and never changed

File: lib/music/MusicInterface.js [hunk 19]
```diff
@@ -0,0 +1,219 @@
...
+    }
+
+    /**
+	 * Makes the bot leave the voiceChannel it is currently in and set the playing status to false
+     * @since 2.0.1
+	 * @param {boolean} [force=true] If the bot should force leave the curreny channel
+	 * @returns {this}
+	 */
+    async leave(force = true) {
+        if (this.playing || this.player || force) this.player.stop();
```
Comment (MrJacz) [inline lib/music/MusicInterface.js]: leave means it leaves the voice channel normally so just do `this.stop(true);`

File: lib/music/MusicInterface.js [hunk 20]
```diff
@@ -0,0 +1,219 @@
...
+        return this;
+    }
+
+    /**
+	 * Starts the player and checks if all criteria is met otherwise it will throw the warning but if it works it will start the player
+     * @since 2.0.1
+	 * @returns {Player}
+	 */
+    async play() {
+        const check = !this.voiceChannel ? "Well, Well, Well, It seems theres no where to play the sick tunes" : !this.player ? "Well it seems I cant find the cable for my 🎧" : this.queue.length === 0 ? "I have no records to play so please provide some" : null;
```
Comment (MrJacz) [inline lib/music/MusicInterface.js]: no, this is wrong.
These checks should be in the play command the the way you're handling this is idiotic

File: lib/music/MusicInterface.js [hunk 21]
```diff
@@ -0,0 +1,219 @@
...
+        if (check) throw check;
+
+        const song = this.queue[0];
```
Comment (MrJacz) [inline lib/music/MusicInterface.js]: this can be `const [song] = this.queue;`

File: lib/music/MusicInterface.js [hunk 22]
```diff
@@ -0,0 +1,219 @@
...
+        this.paused = false;
+        return this;
+    }
+
+    /**
+	 * Depending on the current state it will either pause or resume
+     * @since 2.0.1
+	 * @returns {this}
+	 */
+    smartPlay() {
```
Comment (MrJacz) [inline lib/music/MusicInterface.js]: This seems pointless, should be in play command

File: lib/music/MusicInterface.js [hunk 23]
```diff
@@ -0,0 +1,219 @@
...
+    prune() {
+        this.queue = [];
+        return this;
+    }
+
+    /**
+	 * Destroys the whole MusicInterface by setting everything to null
+     * @since 2.0.1
+	 */
+    async destroy() {
```
Comment (MrJacz) [inline lib/music/MusicInterface.js]: The way the system is meant to be made as it won't work as you think

File: lib/music/MusicInterface.js [hunk 24]
```diff
@@ -0,0 +1,219 @@
...
+        this.queue = null;
+        this.playing = null;
+        this.textChannel = null;
+        this.volume = null;
+
+        await this.leave();
+        this.client.music.delete(this.guild.id);
+    }
+
+    toJSON() {
```
Comment (MrJacz) [inline lib/music/MusicInterface.js]: Pointless toJSON method

File: lib/music/MusicManager.js [hunk 25]
```diff
@@ -0,0 +1,38 @@
...
```
Comment (MrJacz) [inline lib/music/MusicManager.js]: Still needs to be deleted, we dont need it

File: lib/music/Song.js [hunk 26]
```diff
@@ -0,0 +1,46 @@
...
+
+    get friendlyDuration() {
+        return showSeconds(this.duration);
+    }
+
+    toString() {
+        return `${this.title} (${this.url})`;
+    }
+
+    toJSON() {
```
Comment (MrJacz) [inline lib/music/Song.js]: useless toJSON

File: structures/PenguClient.js [hunk 27]
```diff
@@ -1,14 +1,16 @@
...
     constructor(options) { //eslint-disable-line
         super(options);
         this.config = require("../config.json");
         this.functions = require("../utils/functions.js");
-        this.lavalink = null;
-        this.queue = new Map();
         this.log = require("../utils/log.js");
+
+        this.lavalink = require("../lib/music/LavalinkClient");
+        this.music = new MusicManager(this);
```
Comment (MrJacz) [inline structures/PenguClient.js]: Not needed

File: lib/music/MusicInterface.js (hunk 22 above)
Comment (QuantumlyTangled) [reply lib/music/MusicInterface.js]: Well the naming is wrong. It is mainly smart pause

File: lib/music/Song.js (hunk 26 above)
Comment (QuantumlyTangled) [reply lib/music/Song.js]: Non of these?

File: structures/PenguClient.js (hunk 27 above)
Comment (QuantumlyTangled) [reply structures/PenguClient.js]: Non of these?

File: lib/music/MusicManager.js (hunk 17 above)
Comment (MrJacz) [reply lib/music/MusicManager.js]: okay? that isnt what im saying

File: structures/PenguClient.js (hunk 27 above)
Comment (MrJacz) [reply structures/PenguClient.js]: the review is for the line not the block, only music

**Comment by MrJacz** [comment]
@QuantumlyTangled could you rebase rewrite into this branch and resolve conflicts so i can review the PR

File: extendables/music.js (hunk 15 above)
Comment (QuantumlyTangled) [reply extendables/music.js]: Done

File: extendables/player.js (hunk 16 above)
Comment (QuantumlyTangled) [reply extendables/player.js]: After furter discussion this was removed but music is an extendable

**Comment by TheOnlyGhostwolf** [comment]
Zero two best waifu

**Comment by Kashalls** [comment]
Shrug

File: app.js [hunk 28]
```diff
@@ -1,8 +1,9 @@
-const client = require("./structures/PenguClient");
+const client = require("./lib/structures/PenguClient");
 
 client.login(client.config.main.token);
 
-process.on("unhandledRejection", error => console.log(`unhandledRejection:\n${error.stack}`))
+process
+    .on("unhandledRejection", error => console.log(`unhandledRejection:\n${error.stack}`))
     .on("uncaughtException", error => console.log(`uncaughtException:\n${error.stack}`))
```
Comment (MrJacz) [inline app.js]: shouldn't be listening to this

File: arguments/songname.js [hunk 29]
```diff
@@ -0,0 +1,67 @@
...
+            const selection = await msg.awaitReply([`🎵 | **Select a Song - PenguBot**\n`,
+                `${options.map(o => `➡ \`${++index}\` ${o.info.title} - ${o.info.author} (${this.client.functions.friendlyTime(o.info.length)})`).join("\n")}`,
+                `\n${msg.author}, Please select an option by replying from range \`1-5\` to add it to the queue.`], 20000);
+            try {
+                const vid = parseInt(selection);
+                if (vid < 0 || vid > 4) return await msg.send(`${msg.author}, <:penguError:435712890884849664> Invalid Option, select from \`1-5\`. Cancelled request.`);
+                results.push(songsData[vid - 1]);
+            } catch (e) {
+                try {
+                    return await msg.send(`${msg.author}, <:penguError:435712890884849664> No options selected, cancelled request.`);
```
Comment (MrJacz) [inline arguments/songname.js]: use throw without msg.send and dont await return

File: arguments/songname.js [hunk 30]
```diff
@@ -0,0 +1,67 @@
...
+    }
+
+    getType(song) {
+        const ytPlaylist = /(\?|\&)list=(.*)/.exec(song); // eslint-disable-line
+        const ytsmall = /https:\/\/?(www\.)?youtu\.be\/?(.*)/.exec(url);
+        const ytlong = /https:\/\/?(www\.)?youtube\.com\/watch\?v=?(.*)/.exec(url);
+        const soundcloud = /https:\/\/soundcloud\.com\/.*/.exec(url); // eslint-disable-line
+        const scPlaylist = /https:\/\/?soundcloud.com\/.*\/.*\/.*/.exec(song);
+
+        if (ytPlaylist) return "ytplaylist";
```
Comment (MrJacz) [inline arguments/songname.js]: i dislike this idea completely as its unnecessary

File: commands/Music/clearqueue.js [hunk 31]
```diff
@@ -0,0 +1,32 @@
+const { Command } = require("klasa");
+
+module.exports = class extends Command {
+
+    constructor(...args) {
+        super(...args, {
+            runIn: ["text"],
+            cooldown: 10,
+            aliases: ["cq", "prunequeue"],
+            permissionLevel: 0,
```
Comment (MrJacz) [inline commands/Music/clearqueue.js]: pointless

File: commands/Music/clearqueue.js [hunk 32]
```diff
@@ -0,0 +1,32 @@
...
+            requiredPermissions: ["USE_EXTERNAL_EMOJIS"],
+            description: (msg) => msg.language.get("COMMAND_CLEARQUEUE_DESCRIPTION"),
+            extendedHelp: "No extended help available."
+        });
+        this.Music = true;
```
Comment (MrJacz) [inline commands/Music/clearqueue.js]: shouldn't start with a capital and should be called like requireMusic or something like that

File: commands/Music/clearqueue.js [hunk 33]
```diff
@@ -0,0 +1,32 @@
...
+    }
+
+    async run(msg) {
+        const music = msg.guild.music();
```
Comment (MrJacz) [inline commands/Music/clearqueue.js]: music isnt a function and shouldnt be

File: lib/structures/PenguClient.js [hunk 34]
```diff
@@ -0,0 +1,54 @@
...
+class PenguClient extends Client {
+
+    constructor(options) {
+        super({ ...options, permissionLevels });
+        this.config = require("../../config.json");
+        this.functions = require("../Util/Functions.js");
+        this.idiotic = new IdioticAPI(this.config.keys.idiotic, { dev: true });
+
+        this.player = require("../music/LavalinkClient.js");
+        this.music = new MusicManager(this);
```
Comment (MrJacz) [inline lib/structures/PenguClient.js]: this should be removed, its unneeded

File: lib/structures/PenguClient.js [hunk 35]
```diff
@@ -0,0 +1,54 @@
...
+
+class PenguClient extends Client {
+
+    constructor(options) {
+        super({ ...options, permissionLevels });
+        this.config = require("../../config.json");
+        this.functions = require("../Util/Functions.js");
+        this.idiotic = new IdioticAPI(this.config.keys.idiotic, { dev: true });
+
+        this.player = require("../music/LavalinkClient.js");
```
Comment (MrJacz) [inline lib/structures/PenguClient.js]: this should be lavalink, its not a player

File: lib/music/Song.js (hunk 26 above)
Comment (MrJacz) [reply lib/music/Song.js]: Hello?

File: lib/music/MusicManager.js [hunk 36]
```diff
@@ -0,0 +1,34 @@
+const { Guild, Collection } = require("discord.js");
```
Comment (MrJacz) [inline lib/music/MusicManager.js]: Useless collection, MusicInterfactes should be added thru a structure.extend and a property

File: lib/music/MusicInterface.js [hunk 37]
```diff
@@ -0,0 +1,263 @@
...
+            return await lyricsFormatter(req);
+        }
+    }
+
+    /**
+	 * A skip mathod which has force enabled by default and if it stays that way it will either shop the player or just shift the queue
+     * @since 2.0.1
+	 * @returns {this}
+	 */
+    skip() {
```
Comment (MrJacz) [inline lib/music/MusicInterface.js]: skip should check if this.player is not null and use `this.player.stop()`

File: lib/music/MusicInterface.js [hunk 38]
```diff
@@ -0,0 +1,263 @@
...
+        this.loop = true;
+        return this;
+    }
+
+    /**
+	 * Depending on the current state it will either turn autoloop on or off
+     * @since 2.0.1
+	 * @returns {this}
+	 */
+    smartLoop() {
```
Comment (MrJacz) [inline lib/music/MusicInterface.js]: this can be shortened

File: lib/music/MusicInterface.js [hunk 39]
```diff
@@ -0,0 +1,263 @@
...
+        return this;
+    }
+
+    /**
+	 * Depending on the current state it will either pause or resume
+     * @since 2.0.1
+	 * @returns {this}
+	 */
+    smartPause() {
+        if (!this.paused) {
```
Comment (MrJacz) [inline lib/music/MusicInterface.js]: same here

File: lib/music/MusicInterface.js (hunk 20 above)
Comment (MrJacz) [reply lib/music/MusicInterface.js]: Hello?

File: lib/Util/Util.js [hunk 40]
```diff
@@ -0,0 +1,83 @@
...
+    }
+
+}
+
+/**
+ * A function to get a object from the genius api
+ * @param {string} path A song to get
+ * @returns {Object}
+ */
+Util.lyricsRequest = async (path) => {
```
Comment (MrJacz) [inline lib/Util/Util.js]: this should be a static method in the class

File: lib/Util/Util.js [hunk 41]
```diff
@@ -0,0 +1,83 @@
...
+
+    return body;
+};
+
+/**
+ * A function that uses cheerio to scrape results from a given url
+ * @param {string} url The URL to scrape lyrics from
+ * @returns {string}
+ */
+Util.scrapeLyrics = async (url) => {
```
Comment (MrJacz) [inline lib/Util/Util.js]: this should be a static method in the class

File: lib/Util/Util.js [hunk 42]
```diff
@@ -0,0 +1,83 @@
...
+    if (!$(".lyrics")) return null;
+    return $(".lyrics").text().trim();
+};
+
+/**
+ * A function to format the result of the lyricsRequest function
+ * @param {Object} req The object that is returnerned from the lyricsRequest function
+ * @returns {Object}
+ */
+Util.lyricsFormatter = async (req) => {
```
Comment (MrJacz) [inline lib/Util/Util.js]: this should be a static method in the class

File: lib/Util/Util.js [hunk 43]
```diff
@@ -0,0 +1,83 @@
+const snekie = require("snekfetch");
+const cheerio = require("cheerio");
+const config = require("../../config"); // eslint-disable-line
+const key = config.keys.music.lyrics;
```
Comment (MrJacz) [inline lib/Util/Util.js]: destructure this

File: lib/Util/Functions.js [hunk 44]
```diff
@@ -73,15 +73,15 @@ const friendlyTime = (duration) => {
...
+	const regexp = /^(http[s]?:\/\/){0,1}(www\.){0,1}[a-zA-Z0-9\.\-]+\.[a-zA-Z]{2,5}[\.]{0,1}/; // eslint-disable-line
     if (!regexp.test(str)) {
         return false;
     } else { return true; }
 };
 
 const clearPatrons = (client) => {
     if (!client.config.main.patreon) return "Not the Patron Bot.";
-    const guilds = client.guilds; // eslint-disable-line
+	const guilds = client.guilds; // eslint-disable-line
```
Comment (MrJacz) [inline lib/Util/Functions.js]: useless variable

File: lib/Util/Functions.js [hunk 45]
```diff
@@ -1,6 +1,6 @@
 // Any functions which will be available with PenguClient
```
Comment (MrJacz) [inline lib/Util/Functions.js]: this should be in util

File: inhibitors/music.js [hunk 46]
```diff
@@ -0,0 +1,22 @@
+const { Inhibitor } = require("klasa");
+
+module.exports = class extends Inhibitor {
+
+    constructor(...args) {
+        super(...args, { spamProtection: true });
+    }
+
+    async run(msg, cmd) {
+        if (cmd.Music !== true) return;
```
Comment (MrJacz) [inline inhibitors/music.js]: why is there a music command class but then a Music property? this is duplicated code and needs to be fixed before the PR is merged. this should check the instance of cmd and if its MusicCommand and requireMusic is true then do the checks

File: extendables/music.js (hunk 15 above)
Comment (MrJacz) [reply extendables/music.js]: not done, you made it a property... you cant do instanceof properties with extendables please use Discord.js#Structures.extend

File: events/ready.js [hunk 47]
```diff
@@ -1,15 +1,28 @@
...
+            rest: this.client.config.restnode,
+            player: Player,
+            regions: {
+                defaultRegion: this.client.config.regions.defaultRegion,
+                defaultRegions: this.client.config.regions.defaultRegions
+            }
         });
-        this.client.setMaxListeners(50);
-        this.client.lavalink = manager;
+        this.client.emit("log", "[MUSIC] Manager hook has been enabled.");
```
Comment (MrJacz) [inline events/ready.js]: this is kinda pointless

File: events/ready.js [hunk 48]
```diff
@@ -1,15 +1,28 @@
...
-module.exports = class extends Event {
+const MusicClient = require("../lib/music/LavalinkClient");
+const Player = require("../lib/music/Player");
+
+module.exports = class Ready extends Event {
 
     async run() {
-        const manager = new PlayerManager(this.client, this.client.config.nodes, {
+        this.bindMusic();
+        this.client.setMaxListeners(50);
```
Comment (MrJacz) [inline events/ready.js]: 🤔

**Comment by MrJacz** [comment]
Eh, I've been away for the weekend and Monday, When I came home and saw this I was quite disappointed. in the current state, this PR is nowhere close to being merged due to duplicated, idiotic and mess of a code base and some another 24 requested changes. While @QuantumlyTangled has been informed on a lot of details on how this should be done it has been changed in ways it should've not been hopefully this PR does get merged soon so V2 of PenguBot can be live but in the current state this PR cannot be merged into the rewrite branch.

File: commands/Music/clearqueue.js (hunk 33 above)
Comment (AdityaTD) [reply commands/Music/clearqueue.js]: Well, that's the thing, Quantum made it this way so I've been using it like this ever since.

Comment (MrJacz) [reply commands/Music/clearqueue.js]: its a horrible idea and should not be done like this

**Comment by AdityaTD** [comment]
I do agree with @MrJacz but due to the limited amount of time I have every day and the regular tests at school, I'm unable to work on this PR myself, It'd be helpful if you can commit some changes to this branch along with Quantum so we can finish this before June. My holidays start from 1st June so I really can't do much before that. Thanks for the review.

**Comment by Kashalls** [comment]
The argument Songname has a complexity of 22 and is fricken hard to read. Please clean.

**Comment by MrJacz** [comment]
closing due to me creating a new pr tomorrow.

**Pull request closed without merging**